Add support for ModernFortran package

jwortmann opened the pull request:
This assigns the Fortran icon to the `source.fortran` base scope, which is used by an alternative syntax package https://packagecontrol.io/packages/ModernFortran.

braver commented:
It might’ve been better if Modern Fortran would also use the same root scope. But it’s not that important and this change is good. So thanks, let’s ship this!

jwortmann commented:
I think the correct practice to distinguish free-form and fixed-form Fortran source code via the base scope would be
- `source.fortran.free-form`
- `source.fortran.fixed-form`

The term "modern" Fortran isn't really an official term afaik, even though both packages seem to use it. In my opinion the base scopes used by the https://packagecontrol.io/packages/Fortran package are suboptimal, but that package seems to be unmaintained for 5 years or so anyway.

Anyways, thanks for merging.

braver commented:
Yes, that would be correct and allow other packages to transparently share features across bot sub-languages. Anyway, not always something that can be fixed, and not a super big deal.
